src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ProcessLinkerXmlBase.cs

```diff
@@ -191,6 +193,8 @@ protected virtual void ProcessTypes(ModuleDesc assembly, XPathNavigator nav, boo
                 // TODO: Process exported types
 
                 // TODO: Semantics differ and xml format is cecil specific, therefore they are discrepancies on things like nested types
+                // for now just hack replacing / for + to support basic resolving of nested types
+                fullname = fullname.Replace("/", "+");
```

Comment:
Could you file an issue on this? Looks like something we would want to fix for 7.0.

Reply:
Filled https://github.com/dotnet/runtime/issues/73083